.github/workflows/artifacts_CI.yml

```diff
@@ -0,0 +1,112 @@
+name: Artifacts CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    tags: ["*"]
+    paths:
+      - "ray_julia_jll/Artifacts.toml"
+  pull_request:
+    paths:
+      - "ray_julia_jll/Artifacts.toml"
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on:
+      labels: ubuntu-20.04-16core
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - "1.8"  # Version used with MVP
+          - "1.9"  # Latest release
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+      env:
+        ray_julia_jll_dir: ./ray_julia_jll
+        ray_dir: ./ray_julia_jll/deps/ray
+      steps:
+      - uses: actions/checkout@v3
+      - name: Clone Ray repository
+        uses: actions/checkout@v3
+        with:
+          repository: beacon-biosignals/ray
+          path: ${{ env.ray_dir }}
+      - name: Determine Ray commit
+        id: ray-commit
+        run: |
+           cd ${{ env.ray_dir }}
+           head_sha="$(git rev-parse HEAD)"
+           echo "head_sha=$head_sha" | tee -a "$GITHUB_OUTPUT"
+        working-directory: ${{ env.ray_julia_jll-directory }}
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+      # TODO: We shouldn't require separate caches per Julia version but allow cache restores to work
+      # https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+      # https://github.com/beacon-biosignals/Ray.jl/issues/63
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        id: build-cache
+        with:
+          key: build-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
+          path: ~/.cache
+          restore-keys: |
+            build-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
+            build-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
+
+      # Based upon:
+      # https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#building-ray-on-linux-macos-full
+      - name: Build Ray CLI
```

Comment:
Should we also refactor this to just pull our wheels down?

Reply:
Yes we should

Reply:
I'll follow up on this tomorrow

Reply:
https://github.com/beacon-biosignals/Ray.jl/pull/154